### PR TITLE
[Feature] Support for linearly learning rate decay

### PR DESCRIPTION
**Pull request by Sharpiless**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

I tried to re-implement pix2seq bug did not find  linearly learning rate decay in mmdetection

## Modification

add "LinearlyDecayLrUpdaterHook"

test config_file:
```
lr_config = dict(
    policy='LinearAnnealing',
    min_lr=0.001
)
momentum_config = dict(
    policy='LinearAnnealing',
    min_momentum=0.001,
    by_epoch=False
    # anneal_strategy='linear',
)
```
lr curve:
image

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Review by teamwong111 — COMMENTED** [review]
Docstring and unit tests are needed.

File: mmcv/runner/hooks/lr_updater.py [hunk 1]
```diff
@@ -625,31 +625,26 @@ def get_lr(self, runner, base_lr):
 
 
 @HOOKS.register_module()
-class LinearlyDecayLrUpdaterHook(LrUpdaterHook):
+class LinearAnnealingLrUpdaterHook(LrUpdaterHook):
```
Comment (zhouzaida) [inline mmcv/runner/hooks/lr_updater.py:628]: docstring should be added.

File: mmcv/runner/hooks/momentum_updater.py [hunk 2]
```diff
@@ -222,6 +222,30 @@ def get_momentum(self, runner, base_momentum):
                              progress / max_progress)
 
 
+@HOOKS.register_module()
+class LinearAnnealingMomentumUpdaterHook(MomentumUpdaterHook):
+
```
Comment (zhouzaida) [inline mmcv/runner/hooks/momentum_updater.py:227]: docstring should be added.

**Review by teamwong111 — APPROVED** [review]

File: tests/test_runner/test_hooks.py [hunk 3]
```diff
@@ -711,6 +711,83 @@ def test_cosine_runner_hook(multi_optimizers):
     hook.writer.add_scalars.assert_has_calls(calls, any_order=True)
 
 
+@pytest.mark.parametrize('multi_optimziers', (True, False))
```
Comment (zhouzaida) [inline tests/test_runner/test_hooks.py:714]:
```suggestion
@pytest.mark.parametrize('multi_optimizers', (True, False))
```

File: tests/test_runner/test_hooks.py [hunk 4]
```diff
@@ -711,6 +711,83 @@ def test_cosine_runner_hook(multi_optimizers):
...
+    runner.register_hook(IterTimerHook())
+    # add pavi hook
+    hook = PaviLoggerHook(interval=1, add_graph=False, add_last_ckpt=True)
+    runner.register_hook(hook)
+    runner.run([loader], [('train', 1)])
+    shutil.rmtree(runner.work_dir)
+
+    # TODO: use a more elegant way to check values
+    assert hasattr(hook, 'writer')
+    if multi_optimziers:
```
Comment (zhouzaida) [inline tests/test_runner/test_hooks.py:748]:
```suggestion
    if multi_optimizers:
```

**Review by zhouzaida — APPROVED** [review]

**Comment by OpenMMLab-Assistant003** [comment]
Hi @Sharpiless！First of all, we want to express our gratitude for your significant PR in the MMCV project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA

If you have WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤ @Sharpiless

**Pull request merged**